### ISSUE TITLE
Implement employment start and end properties in the users table

### ISSUE THREAD

**Issue by damsleth** [Puzzlepart/did#1211]
### TLDR
Implement `employmentStartDate` and `employmentEndDate` properties on the user objects.

### Description
There are some use cases where having a user start and end date property is useful.

When an employee has their first employment date, e.g. at `2024-03-01`, they are immediately presented with not having forecasted, nor confirmed their hours for the preceeding weeks, even though they were not employed yet.

With an `employmentStartDate` property, the notifications feature (among others) will take this into account, and avoid nagging the user about confirming hours from before their employment began.
